Mention ccls and clangd in README.md

MaskRay opened the pull request:
Whether they work better in lsp-mode is a QoI issue that is unrelated to this PR.

I have checked that both ccls and clangd (built from llvm+clang+extra trunk) work: textDocument/definition.

MaskRay commented:
It's true that I want to promote [ccls](https://github.com/MaskRay/ccls/

joaotavora commented:
So `ccls` is now the default.  I tested it and it works decently well without any extra configuration.  It's a pity that it still has [this abandoned serious cquery bug](https://github.com/cquery-project/cquery/issues/692) though :-(

MaskRay commented:
Implemented `index.onChange: true` in git HEAD. I didn't implement this when I rewrote the pipeline.

https://github.com/MaskRay/ccls/blob/master/src/config.h#L204

This has very bad performance impact and currently the cache file will be saved every time a change is made.

joaotavora commented:
> Implemented index.onChange: true in git HEAD. I didn't implement this when I rewrote the pipeline.

I don't have time to test now, but if I remember correctly (and if you read the bug I linked to) this isn't solved by `index.onChange: true`

MaskRay commented:
I have just added support for empty `cacheDirectory` (default is `".ccls-cache"`)

```javascript
{
  "cacheDirectory": ""
  "index": {
    "onChange": true
  }
}
```

This is on-the-fly indexing without saving cache files.
